client/src/featureform/exceptions.py

```diff
@@ -1,10 +1,10 @@
 
-class InvalidTrainingSetFeatureCategory(Exception):
-    def __init__(self, feature_name, feature_variant, category='ON_DEMAND_CLIENT', message=None):
+class InvalidTrainingSetFeatureComputationMode(Exception):
```

Comment:
Since this is an error the user will see, I wonder if it makes more sense to use `is_on_demand` rather than the `mode`.